### DISCUSSION TITLE
LWFA

### DISCUSSION THREAD

**Discussion by DanRRRR** [epochpic/epoch#657] [General]
Anyone can share any good input deck variants for laser wake field acceleration in 2D? All mine i tried do not look satisfactory

**Comment by Status-Mirror** [comment]
I've got a simple small test deck which I use sometimes. It's loosely based off the parameters in Section 4.1 of [Lifschitz 2009](https://www.sciencedirect.com/science/article/pii/S0021999108005950).

The deck ran for me in just over an hour on 8 cores. I was able to produce this figure of the time-averaged number density:

![wakefield](https://github.com/Warwick-Plasma/epoch/assets/38282483/51ab1a08-6e21-4102-a31c-d736919fea9a)

```
begin:constant
  lambda0 = 0.8e-6
  cell_x = lambda0 /25.0
  cell_y = lambda0 /3.0
  t_fwhm = 30.0e-15
  y_fwhm = 9.0e-6
  t_fw01m = t_fwhm * 1.8226
  wt = t_fwhm * 0.60056
  wy = y_fwhm * 0.60065
end:constant

begin:control
  x_min = 0
  x_max = 68.0 * lambda0
  y_min = -20.0 * lambda0
  y_max = 20.0 * lambda0
  nx = ( x_max - x_min ) / cell_x
  ny = y_max / cell_y
  t_end = 2.9e-12
  stdout_frequency = 1
end:control

begin:boundaries
  bc_x_min = simple_laser
  bc_x_max = open
  bc_y_min = open
  bc_y_max = open
end:boundaries

begin:window
  move_window = T
  window_v_x = 3.0e8
  window_start_time = 160.0e-15
  bc_x_min_after_move = simple_outflow
  bc_x_max_after_move = simple_outflow
end:window

begin:laser
  boundary = x_min
  intensity_w_cm2 = 3.4e18
  lambda = lambda0
  t_profile = gauss ( time , 0.5* t_fw01m , wt )
  profile = gauss (y ,0 , wy )
end:laser

begin:species
  name = Electron
  charge = -1
  mass = 1
  npart = 16 * nx * ny
  density = 7.5e24
  bc_y_max = open
end:species

begin:species
  name = Proton
  charge = 1
  mass = 1836.2
  npart = 4 * nx * ny
  density = 7.5e24
  bc_y_max = open
end:species

begin:output
  name = normal
  dt_snapshot = t_end / 200
  nstep_average = 10 
  grid = always
  ex = always + average
  ey = always + average
  ez = always + average
  number_density = always + average
end:output
```

**Comment by DanRRRR** [comment]
Thanks Stuart, 
Looks like i was failing because tried very different parameter space, was getting always  either dual mode laser, or some strange moire, or no energy gain, etcetc. Will play with this case further, it is good variant for the start.

I am surprised that EPOCH with very inhomogeneous mesh and just 1.5 cells per wavelength in y-direction still gives pretty amicable result.

**Reply by Status-Mirror** [reply]
Oo quite the suspenseful message! I'd be interested to hear what you did to get a roughly 30x speed-boost

**Reply by DanRRRR** [reply]
Your email at Warwich still current?
